### PR TITLE
bluetooth: Refactor libc sockets and add BluetoothMonitorSocket

### PR DESCRIPTION
**Pull request by antoniovazquezblanco**
**Checklist:**

-   [x] If you are new to Scapy: I have checked [CONTRIBUTING.md](https://github.com/secdev/scapy/blob/master/CONTRIBUTING.md) (esp. section submitting-pull-requests)
-   [x] I squashed commits belonging together
-   [x] I added unit tests or explained why they are not relevant
-   [x] I executed the regression tests (using `cd test && ./run_tests` or `tox`)
-   [x] If the PR is still not finished, please create a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/)

Implement a Bluetooth Monitor Socket. Refactor bluetooth libc sockets in the process... I am wondering what testing scope is expected. Let me know what you may consider adecuate...

Thanks

### CODE REVIEW

**Comment by polybassa** [comment]
LGTM

**Review by polybassa — DISMISSED** [review]

**Review by gpotter2 — CHANGES_REQUESTED** [review]
Thanks a lot for the PR !
We don't really have tests for the other bluetooth sockets, so we won't require you to add some. However we have nice bluetooth doc that needs to be slightly updated, to account for the new socket that you added: https://github.com/secdev/scapy/blob/master/doc/scapy/layers/bluetooth.rst Would you mind doing that?

**Comment by antoniovazquezblanco** [comment]
Added documentation for the new socket and also the existing undocumented user socket. I did kind of my best to explain it but I find the lack of information from BlueZ kind of difficult to overcome.

Please, let me know if you find it lacking or if you expect more from it.

Thank you!

File: doc/scapy/layers/bluetooth.rst [hunk 1]
```diff
@@ -71,12 +71,28 @@ There are multiple protocols available for Bluetooth through ``AF_BLUETOOTH``
 sockets:
 
 Host-controller interface (HCI) ``BTPROTO_HCI``
-  Scapy class: ``BluetoothHCISocket``
```
Comment (gpotter2) [inline doc/scapy/layers/bluetooth.rst:74]: Leave this? and eventually move your new description in this?
I'd also add that `BluetoothMonitorSocket` and `BluetoothUserSocket` are essentially subtypes of HCI sockets (if I understand this correctly).

Comment (antoniovazquezblanco) [reply doc/scapy/layers/bluetooth.rst:74]: Sorry for the delay, I've been AFK for a couple of days...

I wanted to provide a generic description for the BPROTO_HCI interface so I moved BluetoothHCISocket down...
I had my doubts on how to order and clasify the sockets as BluetoothMonitorSocket is not purely HCI but finally, I've decided to document from more generic towards more specific and decided that the BluetoothUserSocket should go first as it is the "recommended" one...

Do you still believe I should leave BluetoothHCISocket first? Thanks!

Comment (gpotter2) [reply doc/scapy/layers/bluetooth.rst:74]: Sorry I might be confusing, I can't really visualise REST in my head.

Does the doc look formatted like this:

- HCI
    - monitor
    - User
    - HCI
- L2PCAP

If so it's all good.

Comment (antoniovazquezblanco) [reply doc/scapy/layers/bluetooth.rst:74]: That is correct! Resolving! :D

**Review by gpotter2 — APPROVED** [review]

**Comment by antoniovazquezblanco** [comment]
Thank you! :)

**Pull request merged**